File: python/raindrops/raindrops.py
import itertools

def convert(number):
    if number == 1:
        return '1'
    
    RAIN_DROPS_MAPPING = {
        3: 'Pling',
        5: 'Plang',
        7: 'Plong'
    }

    primes = [1, number]
    primes += list(filter(lambda x: number % x == 0, range(2,number)))
    primes.sort()

    mapping_list = list(map(lambda e: RAIN_DROPS_MAPPING[e], filter(lambda x: x in RAIN_DROPS_MAPPING.keys(), primes)))
    if len(mapping_list) == 0:
        return str(number)
    return ''.join(mapping_list)

File: ruby/raindrops/raindrops.rb
module Raindrops
  def self.convert(num)
    mapping_data = {'3'=>'Pling', '5'=>'Plang', '7'=>'Plong'}
    num_factors = (1..num).select { |e| num%e == 0 }.select { |t| mapping_data.keys.include? t.to_s }
    return num.to_s if num_factors.empty?
    num_factors.map { |e| mapping_data[e.to_s] }.join
  end
end

module BookKeeping
    VERSION = 3
end